src/web3-utils.js

```diff
@@ -125,6 +128,54 @@ export async function getIsContractAccount(web3, account) {
   }
 }
 
+const gasPriceApi = 'https://ethgasstation.info/json/ethgasAPI.json'
+export async function getGasPrice({
+  mainnet: { safeMinimum = '3', disableEstimate } = {},
+} = {}) {
+  if (network.type !== 'main') {
+    // Hardcode 10 for non-mainnet networks
+    return toWei('10', 'gwei')
+  }
+
+  // Mainnet and gas estimation not disabled
+  if (!disableEstimate) {
+    const safeMinimumInWei = toWei(safeMinimum, 'gwei')
+    let priceInWei
+
+    try {
+      const response = await fetch(gasPriceApi, {
+        method: 'GET',
+        mode: 'cors',
+        cache: 'no-cache',
+      })
+      const jsonResponse = await response.json()
+      // Note that all prices from ethgasstation need to be divided by 10 to be in gwei.
+      // The response contains a list of suggested gas prices from 2-120 in
+      // gasPriceRange, so 40 is "slightly higher than the recommended price".
+      const fasterPrice = (jsonResponse.gasPriceRange[40] || 0) / 10
```

Comment:
Should we also `parseInt()` the two values coming from `jsonResponse`? Just in case it returns something unexpected one day.

Reply:
Fixed in 404c31e42eb2b04d249f17879aa0f79dfbe80b1c; broke it out to handle NaNs